### PR TITLE
Fix named prepared statement polyfill

### PR DESCRIPTION
**Pull request by HLeithner**
If you have multiple namedprepared parameters with the same substring in one section of a query the prepared statement polyfill fails.

### Summary of Changes
Only replace the first occur of a found named parameter.

### Testing Instructions
Create a query like this:
```php
$db = JFactory::getDbo();

$query = $db->getQuery(true);

$query->select('id')->from('#__content')->whereIn('id', [1,2,3,4,5,6,7,8,9,10,11]);

$db->setQuery($query);

$list = $db->loadObjectList();
```

### CODE REVIEW

File: Tests/DatabaseQueryTest.php [hunk 1]
```diff
@@ -1818,10 +1818,13 @@ public function testWhere()
 	 */
 	public function testWhereIn()
 	{
-		$this->instance->whereIn('id', [1, 2, 3]);
+		$this->instance->whereIn('id', [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]);
```
Comment (wilsonge) [inline Tests/DatabaseQueryTest.php]: This test is doing nothing - this part is being correctly generated already. please see https://github.com/joomla-framework/database/pull/172 for the correct tests

File: src/Mysqli/MysqliStatement.php [hunk 2]
```diff
@@ -321,7 +326,7 @@ public function bindParam($parameter, &$variable, $dataType = ParameterType::STR
 	private function bindValues(array $values)
 	{
 		$params = [];
-		$types  = str_repeat('s', \count($values));
```
Comment (wilsonge) [inline src/Mysqli/MysqliStatement.php]: This is an incorrect change

File: src/Mysqli/MysqliStatement.php [hunk 3]
```diff
@@ -467,7 +472,7 @@ public function execute($parameters = null)
 		{
 			$this->statement->store_result();
 
-			$this->rowBindedValues = array_fill(0, \count($this->columnNames), null);
+			$this->rowBindedValues = array_fill(0, count($this->columnNames), null);
```
Comment (wilsonge) [inline src/Mysqli/MysqliStatement.php]: so is this

File: Tests/DatabaseQueryTest.php (hunk 1 above)
Comment (HLeithner) [reply Tests/DatabaseQueryTest.php]: I know I wanted to test first with travis

**Comment by mbabker** [comment]
Please merge the `2.0-dev` branch in here to get the tests George added.

**Comment by wilsonge** [comment]
I wrote some extra local tests (which I'll PR in a bit) and I think this is working as expected. Merging and we can amend as required in the future with more tests

**Pull request merged**